builders/tooling/oracles/phala.md

```diff
@@ -0,0 +1,201 @@
+
+Once you're finished editing, your `feeder.ts` file should resemble the below:
+
+??? code "feeder.ts"
+
+    ```solidity
+    --8<-- 'code/builders/tooling/oracles/phala/feeder.ts'
+    ```
+
+### Build and Test {: #build-and-test }
+
+Run the following commands to build and test the project:
+
+```bash
+yarn build
+```
+
+```bash
+yarn test
+```
+
+If everything was successful, you'll see output like the following:
+
+--8<-- 'code/builders/tooling/oracles/phala/terminal/build.md'
+
+### Deploy {: #deploy }
+
+To deploy your aggregator contract for the specified asset/base pair to your EVM Appchain, use the following command:
+
+```bash
+yarn deploy
+```
+
+You'll get a transaction status as well as a contract address. Copy this contract address, as you'll need to refer to it in the following steps. 
+
+--8<-- 'code/builders/tooling/oracles/phala/terminal/deploy.md'
+
+### Access Aggregator Contract {: #access-aggregator-contract }
+
+Next, this tutorial will demonstrate interacting with the newly deployed aggregator contract. Make sure that your MetaMask wallet is connected to your EVM Appchain. You can add your Appchain to your MetaMask by pressing **Add to MetaMask** on your dashboard on [apps.tanssi.network](https://apps.tanssi.network/){target=\_blank}.
+
+Paste the [aggregator contract](https://github.com/smartcontractkit/chainlink/blob/develop/contracts/src/v0.8/shared/interfaces/AggregatorV3Interface.sol){target=\_blank} into a new file in Remix and compile it. 
+
+Then, take the following steps:
+
+1. Head to the **Deploy and Run Transactions** tab
+2. Set the **ENVIRONMENT** to **Injected Provider -- MetaMask** and ensure that your MetaMask is on the network of your EVM Appchain. You can verify the EVM chain ID in Remix if you are unsure
+3. Select the `AggregatorV3Interface` contract from the **CONTRACT** dropdown
+4. Enter the data feed contract address corresponding to your desired asset pair that was returned on the command line in the prior section in the **At Address** field and click the **At Address** button
+
+![Access aggregator contract](/images/builders/tooling/oracles/phala/phala-5.webp)
+
+Expand the `AggregatorV3Interface` contract to reveal the available functions and click `latestRoundData` to see the most recent price data for the asset pair. You should see `0` values for all. This is because your aggregator contract has been deployed, but it hasn't yet fetched price data. You can fix this with a quick price feed update. 
```

Comment:
```suggestion
Expand the **AggregatorV3Interface** contract to reveal the available functions and click **latestRoundData** to see the most recent price data for the asset pair. You should see `0` values for all. This is because your aggregator contract has been deployed, but it hasn't yet fetched price data. You can fix this with a quick price feed update. 
```